### DISCUSSION TITLE
Edit a metadata key name without deleting the whole item

### DISCUSSION THREAD

**Discussion by gamma-delta** [godotengine/godot-proposals#7624] [Editor]
Hello hello. This is my first time interacting with Godot other than as a user so please let me know if I'm doing anything wrong.

I'm writing some simple utility AI, and for prototyping I'd like to be able to plop a bunch of advertisers in the scene and have them broadcast the need they meet. For the time being I've decided to put this in a `PackedStringArray` metadata. (I'll move to tons of scenes with tons of scripts once I get the basics working.)

The problem is; when I make a typo in the metadata key, I can't edit the key without deleting the whole item and starting over.

![image](https://github.com/godotengine/godot-proposals/assets/29877714/52611d96-fde7-4189-8135-d95af8d4fdbd)

This is *supposed* to say `advertisements`, but I mistyped.

Given that it's only 1 long I can pretty easily just retype the whole item, and I'm not going to be using metadata for forever, but it seems like a big shortcoming of the engine to me to not be able to edit that key. (Also, the fact that it automatically converts the case of the name like it does with other editor `[Export]`s.)

**Comment by Mickeon** [comment]
You cannot rename metadata entries in code. But this is a Metadata editor oversight, as it feels like you *should be able to*.

Fortunately for you, you may open the .tscn/.tres file of this Scene/Resource with a Text editor of your choice, and take a look inside.
You may find a property called `metadata/advertisement` inside, that you're free to rename without much fuss. Something like this:
![image](https://github.com/godotengine/godot-proposals/assets/66727710/0ef53d15-0466-4b27-b203-9eed4f25df36)

**Comment by Calinou** [comment]
> (Also, the fact that it automatically converts the case of the name like it does with other editor [Export]s.)

If Dictionary values in the inspector behave this way, I'd expect metadata properties to behave the exact same way.
